cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMediumTypeCode.java

```diff
@@ -0,0 +1,34 @@
+package no.unit.nva.cristin.mapper;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import nva.commons.core.JacocoGenerated;
+
+public enum CristinMediumTypeCode {
+    JOURNAL("TIDSSKRIFT"),
+    TV("TV"),
+    PROFESSIONAL_JOURNAL("FAGBLAD"),
+    INTERNET("INTERNETT"),
+    NEWSPAPER("AVIS"),
+    RADIO("RADIO");
+
+    private final String value;
+
+    CristinMediumTypeCode(String value) {
+        this.value = value;
+    }
+
+    public static CristinMediumTypeCode fromValue(String value) {
+        for (CristinMediumTypeCode type : CristinMediumTypeCode.values()) {
+            if (type.getValue().equalsIgnoreCase(value)) {
+                return type;
+            }
+        }
+        throw new IllegalArgumentException();
```

Comment:
Generally, I'd use a stream of values() and SingletonCollector, but I don't think it's important

Reply:
Fikset